src/mlpack/methods/ann/layer/hardshrink.hpp

```diff
@@ -0,0 +1,218 @@
+/**
+ * @file hardshrink.hpp
+ * @author Lakshya Ojha
+ *
+ * Same as soft thresholding, if its amplitude is smaller than a predefined
+ * threshold, it will be set to zero (kill), otherwise it will be kept 
+ * unchanged.
+ * In order to promote sparsity and to improve the approximation, the hard
+ * thresholding method is used as an alternative.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_HARDSHRINK_HPP
+#define MLPACK_METHODS_ANN_LAYER_HARDSHRINK_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artifical Neural Network. */ {
+
+ /**
+ * Hard Shrink operator is defined as,
+ * @f{eqnarray*}{
+ * f(x) &=& \left\{
+ *   \begin{array}{lr}
+ *     x  & : x >  lambda \\
+ *     x  & : x < -lambda \\
+ *     0  & : otherwise
+ *   \end{array} \\
+ * \right.
+ * f'(x) &=& \left\{
+ *   \begin{array}{lr}
+ *     1 & : x >  lambda \\
+ *     1 & : x < -lambda \\
+ *     0 & : otherwise
+ *   \end{array}
+ * \right.
+ * @f}
+ * lambda is set to 0.5 by default.
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class HardShrink
+{
+ public:
+/**
+   * Create HardShrink object using specified hyperparameter lambda.
+   * 
+   * @param lambda is calculated by multiplying the
```

Comment:
Picky style comment, the first word after the parameter should use upper casing.

Reply:
`* @param lambda Is calculated by multiplying the` 
Is this the correct way to do so?

Reply:
That looks good, yes.

Reply:
Thanks for the quick reply.
I have done the changes as above.